tests/test_random.py

```diff
@@ -772,6 +772,105 @@ def test_poisson():
+        p = galsim.PoissonDeviate(testseed, mean=mean)
+        p2 = p.duplicate()
+        p3 = galsim.PoissonDeviate(p.serialize(), mean=mean)
+        testResult = (p(), p(), p())
+        testResult2 = (p2(), p2(), p2())
+        testResult3 = (p3(), p3(), p3())
+        np.testing.assert_allclose(
+                testResult2, testResult, rtol=1.e-8,
+                err_msg='PoissonDeviate.duplicate not equivalent for mean=%s'%mean)
+        np.testing.assert_allclose(
+                testResult3, testResult, rtol=1.e-8,
+                err_msg='PoissonDeviate from serialize not equivalent for mean=%s'%mean)
+
+        # Check that the mean and variance come out right
+        vals = [p() for i in range(nvals)]
+        mu = np.mean(vals)
+        var = np.var(vals)
+        print('mean = ',mu,'  true mean = ',mean)
+        print('var = ',var,'   true var = ',mean)
+        np.testing.assert_allclose(mu, mean, rtol=1.e-5,
+                err_msg='Wrong mean from PoissonDeviate with mean=%s'%mean)
+        np.testing.assert_allclose(var, mean, rtol=rtol_var,
+                err_msg='Wrong variance from PoissonDeviate with mean=%s'%mean)
+
+        # Check that two connected poisson deviates work correctly together.
+        p2 = galsim.PoissonDeviate(testseed, mean=mean)
+        p.reset(p2)
+        testResult2 = (p(), p(), p2())
+        np.testing.assert_array_equal(
+                testResult2, testResult,
+                err_msg='Wrong poisson random number sequence generated using two pds')
+        p.seed(testseed)
+        p2.clearCache()
+        testResult2 = (p2(), p2(), p())
+        np.testing.assert_array_equal(
+                testResult2, testResult,
+                err_msg='Wrong poisson random number sequence generated using two pds after seed')
+
+        # Test filling an image
+        p.seed(testseed)
+        testimage = galsim.ImageD(np.zeros((3, 1)))
+        testimage.addNoise(galsim.DeviateNoise(p))
+        np.testing.assert_array_equal(
+                testimage.array.flatten(), testResult,
+                err_msg='Wrong poisson random number sequence generated when applied to image.')
+
+        # The PoissonNoise version also subtracts off the mean value
+        rng = galsim.BaseDeviate(testseed)
+        pn = galsim.PoissonNoise(rng, sky_level=mean)
+        testimage.fill(0)
+        testimage.addNoise(pn)
+        np.testing.assert_array_equal(
+                testimage.array.flatten(), np.array(testResult)-mean,
+                err_msg='Wrong poisson random number sequence generated using PoissonNoise')
+
+        # Check PoissonNoise variance:
+        np.testing.assert_allclose(
+                pn.getVariance(), mean, rtol=1.e-8,
+                err_msg="PoissonNoise getVariance returns wrong variance")
+        np.testing.assert_allclose(
+                pn.getSkyLevel(), mean, rtol=1.e-8,
+                err_msg="PoissonNoise getSkyLevel returns wrong value")
+        continue
```

Comment:
Good catch.  No, I didn't.  I'd added that when debugging something and forgot to remove it.